transfer-family.tf

```diff
@@ -1,10 +1,19 @@
 resource "aws_transfer_server" "default" {
-  identity_provider_type    = "SERVICE_MANAGED"
-  protocols                 = ["SFTP"]
-  endpoint_type             = "PUBLIC"
-  force_destroy             = true
-  security_policy_name      = var.security_policy_name
-  logging_role              = join("", aws_iam_role.logging[*].arn)
+  identity_provider_type = "SERVICE_MANAGED"
+  protocols              = ["SFTP"]
+  endpoint_type          = var.endpoint_type
+  force_destroy          = true
+  security_policy_name   = var.security_policy_name
+  logging_role           = join("", aws_iam_role.logging[*].arn)
+
+  dynamic "endpoint_details" {
+    for_each = var.endpoint_type == "VPC" ? [1] : []
+    content {
+      address_allocation_ids = var.address_allocation_ids
+      subnet_ids             = var.public_subnet_ids
+      vpc_id                 = var.vpc_id
```

Comment:
wasn't possible to pass the security group here instead of using the null resource down below?